tests/config.rs

```diff
@@ -24,105 +21,6 @@ fn test_config_subcommand_produces_valid_toml_configuration_dump() {
     toml::from_str::<Config>(std::str::from_utf8(output).unwrap()).expect("Invalid configuration!");
 }
 
-#[test]
-fn test_overrides_are_reflected_in_configuration_dump() {
-    let mut rng = rand::thread_rng();
-    let randomized_metrics_host = format!("127.0.0.1:{}", rng.gen::<u16>());
-
-    let cmd = Command::cargo_bin("forest-cli")
-        .unwrap()
-        .arg("--rpc")
-        .arg("true")
-        .arg("--token")
-        .arg("Azazello")
-        .arg("--metrics-address")
-        .arg(&randomized_metrics_host)
-        .arg("config")
-        .arg("dump")
-        .assert()
-        .success();
-
-    let output = &cmd.get_output().stdout;
-    let config = toml::from_str::<Config>(std::str::from_utf8(output).unwrap())
-        .expect("Invalid configuration!");
-
-    assert_eq!(
-        config.client.metrics_address,
-        SocketAddr::from_str(&randomized_metrics_host).unwrap()
-    );
-}
-
-#[test]
-fn test_reading_configuration_from_file() {
-    let mut rng = rand::thread_rng();
-
-    let expected_config = Config {
-        client: Client {
-            rpc_token: Some("Azazello".into()),
-            genesis_file: Some("cthulhu".into()),
-            encrypt_keystore: false,
-            metrics_address: SocketAddr::from_str(&format!("127.0.0.1:{}", rng.gen::<u16>()))
-                .unwrap(),
-            ..Client::default()
-        },
-        ..Config::default()
-    };
-
-    let mut config_file = tempfile::Builder::new().tempfile().unwrap();
-    config_file
-        .write_all(toml::to_string(&expected_config).unwrap().as_bytes())
-        .expect("Failed writing configuration!");
-
-    let cmd = Command::cargo_bin("forest-cli")
-        .unwrap()
-        .arg("--rpc")
-        .arg("true")
-        .arg("--token")
-        .arg("Azazello")
-        .arg("--config")
-        .arg(config_file.path())
-        .arg("config")
-        .arg("dump")
-        .assert()
-        .success();
-
-    let output = &cmd.get_output().stdout;
-    let actual_config = toml::from_str::<Config>(std::str::from_utf8(output).unwrap())
-        .expect("Invalid configuration!");
-
-    assert_eq!(expected_config, actual_config);
-}
-
-#[test]
```

Comment:
Any suggestions for how to test this with the daemon?

Reply:
Something like this https://github.com/ChainSafe/forest/blob/main/tests/keystore_tests.rs, but instead of providing the file, provide the environmental variable. Assert some settings (some are getting overridden by the daemon so it may be difficult to have full equality there). What do you think?

Reply:
So, we add something like `--print-config-and-halt` to the daemon?

Reply:
No. Just pass an existing configuration (perhaps generated by the `config default` you suggested), modify a value there, run Forest with it with the environmental variable and check that it picked up the config.

Reply:
Oh, so `config default` should read configuration files? That doesn't seem right to me. Maybe I'm misunderstanding something.

Reply:
No, it should generate configuration feels that are _feedable_ to the daemon.

Reply:
> No, it should generate configuration feels that are _feedable_ to the daemon.

I don't know what this means. You may need to draw me a diagram or something.